test_mq_service.py

```diff
@@ -37,10 +37,12 @@ class MockEventProducer:
     def __init__(self):
         self.event = None
         self.key = None
+        self.headers = None
 
-    def write_event(self, event, key):
+    def write_event(self, event, key, headers):
         self.event = event
         self.key = key
+        self.headers = [(hk, hv.encode("utf-8")) for hk, hv in headers.items()]
```

Comment:
Well, actually we don’t need to replicate the internal logic of [_KafkaEventProducer_](https://github.com/RedHatInsights/insights-host-inventory/blob/71241da25ea6e952b42ac7a20e3f28d36e632c7b/app/queue/egress.py#L16). The transformation is done only because _KafkaProducer_ needs it, but here we just compare the values.

```suggestion
        self.headers = headers
```

I think we don’t even necessarily need the class, but here it’s used on many places, so let’s keep it for now. But if you want to do the same as in the API tests, it’d be awesome, although maybe out of the scope of this pull request.

Also, [_FakeKafkaMessage_](https://github.com/RedHatInsights/insights-host-inventory/blob/71241da25ea6e952b42ac7a20e3f28d36e632c7b/test_mq_service.py#L29) just above is not used at all. Also unrelated, but… remove?